File: app/src/main/java/workshop/akbolatss/xchangesrates/screens/snapshots/SnapshotsView.java
package workshop.akbolatss.xchangesrates.screens.snapshots;

import java.util.List;

import workshop.akbolatss.xchangesrates.base.BaseView;
import workshop.akbolatss.xchangesrates.base.LoadingView;
import workshop.akbolatss.xchangesrates.model.dao.Snapshot;
import workshop.akbolatss.xchangesrates.model.dao.SnapshotInfo;

/**
 * Author: Akbolat Sadvakassov
 * Date: 04.01.2018
 */

public interface SnapshotsView extends BaseView, LoadingView {

    public void onLoadSnapshots(List<Snapshot> snapshotList);

    public void onLoadInfo(SnapshotInfo dataInfo, int pos);

    public void onLoadChart(Snapshot data, int pos);

    public void onSaveNotifiesCount(int count);
}

File: app/src/main/java/workshop/akbolatss/xchangesrates/screens/snapshots/SnapshotsView.kt
package workshop.akbolatss.xchangesrates.screens.snapshots

import workshop.akbolatss.xchangesrates.base.BaseView
import workshop.akbolatss.xchangesrates.model.response.ChartData

/**
 * Author: Akbolat Sadvakassov
 * Date: 04.01.2018
 */

interface SnapshotsView : BaseView {

    /**
     * Loading all snapshots to RecyclerView
     */
    fun loadChartDatas(chartDataList: List<ChartData>)

    /**
     * Loading single snapshot to RV
     */
    fun loadChartData(chartData: ChartData, pos: Int)


    fun onErrorChartItem(pos: Int)
    fun enqueueWorker(it: ChartData)
    fun dequeueWorker(chartData: ChartData)
}
